app/Controller/UsersController.php

```diff
@@ -34,33 +34,50 @@ public function beforeFilter() {
 		$this->Auth->allow('login', 'logout');
 	}
 
-/**
- * view method
- *
- * @param string $id
- * @return void
- * @throws NotFoundException
- */
+	/**
+	 * view method
+	 *
+	 * @param string $id
+	 * @return void
+	 * @throws NotFoundException
+	 */
 	public function view($id = null) {
 		if ("me" == $id) $id = $this->Auth->user('id');
-		if (!$this->_isSiteAdmin() && $this->Auth->user('id') != $id) {
-			throw new NotFoundException(__('Invalid user or not authorised.'));
+		if (!$this->_isSiteAdmin() && ($this->Auth->user('id') != $id)) {
+				throw new NotFoundException(__('Invalid user or not authorised.'));
 		}
 		$this->User->id = $id;
 		$this->User->recursive = 0;
+
 		if (!$this->User->exists()) {
 			throw new NotFoundException(__('Invalid user'));
 		}
 		$this->set('user', $this->User->read(null, $id));
 	}
 
-/**
- * edit method
- *
- * @param string $id
- * @return void
- * @throws NotFoundException
- */
+	public function request_API(){
+		$responsibleAdmin = $this->User->findAdminsResponsibleForUser($this->Auth->user('id'));
```

Comment:
Why not pass $this->Auth->user() here? That way you wouldn't have to retrieve the user again in findAdminsResponsibleForUser()